### PR TITLE
Remove namespace from `linkerd-smi` chart

### PR DESCRIPTION
**Pull request by Pothulapati**
Another part of #6584

This PR
- stops rendering `namespace.yaml` in the `linkerd-smi` chart.
- Adds a new `post-install` hook to add the `linkerd.io/extension`
  label to the namespace after the chart is installed.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: charts/linkerd-smi/templates/namespace-metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+metadata:
+  annotations:
+    {{ include "partials.annotations.created-by" . }}
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "0"
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+  labels:
+    app.kubernetes.io/name: namespace-metadata
+    app.kubernetes.io/part-of: Linkerd
+    app.kubernetes.io/version: {{default .Values.linkerdVersion .Values.cliVersion}}
```
Comment (alpeb) [inline charts/linkerd-smi/templates/namespace-metadata.yaml]: These values aren't defined anywhere

File: charts/linkerd-smi/templates/namespace-metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  name: namespace-metadata
+spec:
+  template:
+    metadata:
+      annotations:
+        {{ include "partials.annotations.created-by" . }}
+      labels:
+        app.kubernetes.io/name: namespace-metadata
+        app.kubernetes.io/part-of: Linkerd
+        app.kubernetes.io/version: {{default .Values.linkerdVersion .Values.cliVersion}}
```
Comment (alpeb) [inline charts/linkerd-smi/templates/namespace-metadata.yaml]: ditto

**Comment by alpeb** [comment]
I think it's fine to drop the "created-by" annotation from the new namespace-metadata template to fix that integration test :wink:

**Review by alpeb — APPROVED** [review]
LGTM :+1: 
I think we might wanna hold merging this until we document the upgrade strategy, and figure a new name and location for a new chart, given this change is not backwards-compatible.

**Review by kleimkuhler — APPROVED** [review]

**Pull request merged**